pkg/condition/conditions.go

```diff
@@ -0,0 +1,61 @@
+package condition
+
+import (
+	gcpv1alpha1 "github.com/openshift/gcp-project-operator/pkg/apis/gcp/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Conditions is a wrapper object for actual Condition functions to allow for easier mocking/testing.
+type Conditions interface {
+	SetCondition(conditions *[]gcpv1alpha1.Condition, conditionType gcpv1alpha1.ConditionType, status corev1.ConditionStatus, reason string, message string)
+}
+
+type ConditionManager struct {
+}
+
+// NewConditionManager returns a ConditionManager object
+func NewConditionManager() Conditions {
+	return &ConditionManager{}
+}
+
+// SetCondition sets a condition on a custom resource's status
+func (c *ConditionManager) SetCondition(conditions *[]gcpv1alpha1.Condition, conditionType gcpv1alpha1.ConditionType, status corev1.ConditionStatus, reason string, message string) {
+	now := metav1.Now()
+	existingCondition := c.FindCondition(*conditions, conditionType)
+	if existingCondition == nil {
+		*conditions = append(
+			*conditions,
+			gcpv1alpha1.Condition{
+				Type:               conditionType,
+				Status:             status,
+				Reason:             reason,
+				Message:            message,
+				LastTransitionTime: now,
+				LastProbeTime:      now,
+			},
+		)
+	} else {
+		if status == corev1.ConditionTrue {
+			existingCondition.Status = "IssueIsOngoing"
+			existingCondition.Reason = reason
+			existingCondition.Message = message
+			existingCondition.LastProbeTime = now
+		} else {
+			existingCondition.Status = "Resolved"
```

Comment:
the type of the status field is corev1.ConditionStatus, I think we shouldn't assign a string to it.

Reply:
Can we use True and False again here(true: ongoing, false: resolved), and use the Reason field to make it visible when the error resolved? (`ReconcileError` and `ReconcileErrorResolved` probably?)

Reply:
That would also make the `SetCondition` function more generic, currently it would be only useful for the error condition (which is the only one we have so far).

Reply:
@NautiluX done, now we're using `reason` field for that.